### ISSUE TITLE
Allow copying of assessments to other course instances (and other courses?)

### ISSUE THREAD

**Issue by mwest1066** [PrairieLearn/PrairieLearn#2949]
At the moment the "Copy assessment" button on the assessment Settings page always makes a copy in the same course assessment. It would be good to allow the user to choose a different course instance (perhaps in a different course?). This could work like the "Copy question" button that allows copying to a different course.

**Comment by nwalters512** [comment]
I'd separate this into two tasks: allow copying between course instances (borderline trivial) and copying between courses (much harder, as you then have to deal with question sharing, or copying all dependent questions, or whatever).

**Comment by mwest1066** [comment]
Agreed on this separation. For copying between courses it would be awesome if we can automatically update questions to handle question sharing. There are quite a few cases of this:
1. best case: source course is sharing the question publicly or with destination course, so we can update to `@source-course/QID`
2. also best case: question was already shared like `@course3/QID` and `course3` is also sharing with destination course, so we can just copy this as is
3. intermediate case: we can't share the question, but we have access to the source and so we can copy it to the destination course
4. another intermediate case: question was already shared like `@course3/QID`, we _don't_ have access to share to destination course, but we _do_ have access to copy the source to destination course
5. bad case: question was shared like `@course3/QID` but we don't have access to share or copy the question to the destination course. Maybe give an error and refuse to copy the assessment?

I think that in cases (1) and (2) above we should just go ahead and copy the assessment. In cases (3) and (4) we might want to warn the user "we are about to also copy 8 questions into <destination>, do you want to proceed?".

It's also possible that someone might want to copy all the questions even if they could be shared. Maybe we should present this as an option?
